apps/api/src/lib/sentry-config.ts

```diff
@@ -0,0 +1,139 @@
+
+// Sentry types for sampling context
+interface SentryTransactionContext {
+  name?: string;
+}
+
+interface SentrySamplingContext {
+  transactionContext?: SentryTransactionContext;
+  name?: string;
+}
+
+// Use Sentry's built-in types for compatibility
+
+// API sampling rate defaults (production-safe values)
+export const API_SENTRY_SAMPLING = {
+  DEFAULT_TRACES_RATE: 0.005,
+  DEFAULT_CRITICAL_RATE: 0.1, // Higher for trading/admin endpoints
+  DEFAULT_API_RATE: 0.01, // Standard API endpoints
+  DEFAULT_DATABASE_RATE: 0.01, // Database-heavy endpoints
+} as const;
+
+// Helper to get sampling rate from env with fallback
+function getSamplingRate(
+  envVar: string | undefined,
+  defaultRate: number,
+): number {
+  return envVar ? parseFloat(envVar) : defaultRate;
+}
+
+// API-specific configuration object
+export interface ApiSentryConfig {
+  dsn: string | undefined;
+  environment: string;
+  enabled: boolean;
+  dbMonitoringEnabled: boolean;
+  sampling: {
+    traces: number;
+    critical: number;
+    api: number;
+    database: number;
+  };
+}
+
+// Create API Sentry config from environment variables
+export function createSentryConfig(): ApiSentryConfig {
+  const dsn = process.env.SENTRY_DSN;
+
+  return {
+    dsn,
+    environment:
+      process.env.SENTRY_ENVIRONMENT || process.env.NODE_ENV || "development",
+    enabled: !!dsn,
+    dbMonitoringEnabled:
+      !!dsn && process.env.ENABLE_SENTRY_DB_MONITORING === "true",
+    sampling: {
+      traces: getSamplingRate(
+        process.env.SENTRY_TRACES_SAMPLE_RATE,
+        API_SENTRY_SAMPLING.DEFAULT_TRACES_RATE,
+      ),
+      critical: getSamplingRate(
+        process.env.SENTRY_CRITICAL_SAMPLE_RATE,
+        API_SENTRY_SAMPLING.DEFAULT_CRITICAL_RATE,
+      ),
+      api: getSamplingRate(
+        process.env.SENTRY_API_SAMPLE_RATE,
+        API_SENTRY_SAMPLING.DEFAULT_API_RATE,
+      ),
+      database: getSamplingRate(
+        process.env.SENTRY_DATABASE_SAMPLE_RATE,
+        API_SENTRY_SAMPLING.DEFAULT_DATABASE_RATE,
+      ),
+    },
+  };
+}
+
+// API-specific transaction sampling logic
+export function createTracesSampler(config: ApiSentryConfig) {
+  return (samplingContext: SentrySamplingContext) => {
+    const transactionName =
+      samplingContext.transactionContext?.name || samplingContext.name;
+
+    // Critical business operations: higher sampling for trading and admin
+    if (
+      transactionName?.includes("/trade") ||
+      transactionName?.includes("/competition") ||
+      transactionName?.includes("/admin/")
+    ) {
+      return config.sampling.critical;
+    }
+
+    // Health checks and metrics: no sampling to reduce noise
+    if (
+      transactionName?.includes("/health") ||
+      transactionName?.includes("/metrics")
+    ) {
+      return 0;
+    }
+
+    // Database-heavy endpoints: reduced sampling due to volume
+    if (
+      transactionName?.includes("/leaderboard") ||
+      transactionName?.includes("/portfolio") ||
```

Comment:
`/portfolio` isnt an endpoint

Reply:
huh interesting. i wonder why? we _used to_ have a `/portfolio` endpoint. i dont think we've ever had `/stats`. perhaps someone is hammering our APIs at non-existent endpoints, and that shows up in sentry?